### PR TITLE
Metric precision

### PR DESCRIPTION
**Pull request by lachhebo**
I have added the possibility to choose an other metric. 

The only metric i have added for now is precision. It is only useful for binary classification for now but it will probably lead to better results in unbalanced datasets. 

We could easily expand it to let users choose others metrics.

### CODE REVIEW

**Comment by Menelau** [comment]
@lachhebo The tests are failing in this PR, by the look at the log from Travis I believe the problem is due to the change in variable names in the DESClustering class. Can you check this problem?

About the metric, maybe we could expand by considering  the classification metrics from sklearn.metrics: https://scikit-learn.org/stable/modules/classes.html#classification-metrics 
I think they have a way to fetch the right performance metric based on its name (as used in the GridSearchCV class), but I need to check how it works.

**Comment by lachhebo** [comment]
Yes, i need to modify the test to because i renamed the argument metric to metric_diversity as we have two kind of metric now. 
Same thing for the attribute accuracy_cluster_ i renamed to performance_cluster_.

Yes, it is a good idea, may be some metrics won't work because in some clusters, some labels are in labels_predicted and not in label_true (y).

**Comment by Menelau** [comment]
It seems that everything is working fine now, including the support of different metrics. The only problem with the code now is that it is violating several PEP8 errors, can you fix these issues? 
After that, I can merge the PR.

**Comment by Menelau** [comment]
All good, Thanks for your contribution!
